README.md

```diff
@@ -0,0 +1,174 @@
+-   Batch, in which the data has no meaningful ordering with respect to time,
+    and the goal is comparing two datasets as a whole.
+
+Additionally, the library implements a kdq-Tree partitioner, for support of the
+kdq-Tree Detection Method. This data structure partitions a given feature space,
+then maintains a count of the number of samples from the given dataset that fall
+into each section of that partition. More details are given in the respective
+module.
+
+A flowchart breaking down these contexts can be found on the ReadTheDocs page under "Choosing a Detector."
+
+# Installation
+
+Create a virtual environment as desired, e.g. `python -m venv ./venv`,
+then:
+
+```python
+# for read-only, install from pypi:
+pip install menelaus
+
+# to allow editing, running tests, generating docs, etc.
+# First, clone the git repo, then:
+cd ./menelaus/
+pip install -e .[dev] 
+```
+
+Menelaus should work with Python 3.8 or higher. 
+
+# Getting Started
+
+Each detector implements the API defined by `menelaus.drift_detector`:
+they have an `update` method which allows new data to be passed, a
+`drift_state` attribute which tells the user whether drift has been
+detected, and a `reset` method (generally called automatically by
+`update`) which clears the `drift_state` along with (usually) some other
+attributes specific to the detector class.
+
+Generally, the workflow for using a detector, given some data, is as
+follows:
+
+```python
+import pandas as pd
+from menelaus.concept_drift import ADWIN
+df = pd.read_csv('example.csv')
+detector = ADWIN()
+for i, row in df.iterrows():
+   detector.update(row['y_predicted'], row['y_true'])
+   if detector.drift_state is not None:
+      print("Drift has occurred!")
+```
+
+For this example, because ADWIN is a concept drift detector, it requires
+both a predicted value (`y_predicted`) and a true value (`y_true`), at
+each update step. Note that this requirement is not true for the
+detectors in other modules. More detailed examples, including code for
+visualizating drift locations, may be found in the ``examples`` directory, as
+stand-alone python scripts. The examples along with output can also be viewed on
+the RTD website.
+
+# Contributing
+Install the library using the `[dev]` option, as above.
+
+- **Testing**
+
+  Unit tests can be run with the command `pytest`. By default, a
+  coverage report with highlighting will be generated in `htmlcov/index.html`.
+  These default settings are specified in `setup.cfg` under `[tool:pytest]`.
+
+- **Documentation**
+
+  HTML documentation can be generated at
+  `menelaus/docs/build/html/index.html` with:
+  ```python
+  cd docs/source
+  sphinx-build . ../build
+  ```
+
+- **Formatting**:
+
+  This project uses `black` and `flake8` for code formatting and
+  linting, respectively. To satisfy these requirements when contributing, you
+  may use them as the linter/formatter in your IDE, or manually run the
+  following from the root directory:
```

Comment:
Checking for my understanding -- so they don't need to do anything about bandit?

Reply:
Added.